engine/access/ingestion/engine.go

```diff
@@ -757,6 +854,17 @@ func (e *Engine) updateLastFullBlockReceivedIndex() {
 		e.metrics.UpdateLastFullBlockHeight(lastFullHeight)
 	}
 
+	// if we've fallen behind by more than a few blocks, try to catch up using execution data
+	if latestFullHeight < finalBlk.Height-defaultExecutionDataCatchupDistance {
```

Comment:
preventing from underflow
```suggestion
	if latestFullHeight + defaultExecutionDataCatchupDistance < finalBlk.Height {
```